hyperspy/signal.py

```diff
@@ -3020,6 +3032,13 @@ def sum(self, axis=None, out=None, rechunk=True):
         """
         if axis is None:
             axis = self.axes_manager.navigation_axes
+        axes = self.axes_manager[axis]
+        if not np.iterable(axes):
+            axes = (axes,)
+        if any([not ax.is_linear for ax in axes]):
+                warnings.warn("You just summed over a non-linear axis. The "
```

Comment:
I am not convinced that this warning is necessary. This method is calculating the sum and not the integration so it should fine? For the integration, there are the `integrate1D` and `integrate_simpson` methods.

Reply:
Yes, I guess we can assume people are smart enough. Summing over a non-linear axis works, but it makes only limited sense. But I've seen that it is e.g. used in the navigator-plot for linescans.

Reply:
My concern here is that this warning will always warn the users regardless if this is correct or not. This is a valid operation, so it should be possible to run it without warning. As usual, it is a good practise to make it clear enough in the docstring and user guide if there is a need for warning/clarification.